Availability set and PlatformFaultDomainCount

Trying to deploy on Azure within the wrong region could have some severe consequences. 
For example, using the West Central US, will lead to errors when using the availability set defaults

### Terraform Version
`Terraform v0.11.3`

### Affected Resource(s)
Please list the resources as a list, for example:
- azurerm_availability_set

### Terraform Configuration Files
```hcl
resource "azurerm_resource_group" "test" {
  name     = "resourceGroup1"
  location = "West US"
}

resource "azurerm_availability_set" "test" {
  name                = "acceptanceTestAvailabilitySet1"
  location            = "${azurerm_resource_group.test.location}"
  resource_group_name = "${azurerm_resource_group.test.name}"

  tags {
    environment = "Production"
  }
}
```

### Debug Output
```
[ERROR] root: eval: *terraform.EvalApplyPost, err: 1 error(s) occurred: azurerm_availability_set.vm: compute.AvailabilitySetsClient#CreateOrUpdate: Failure responding to request: StatusCode=400 -- Original Error: autorest/azure: Service returned an error. Status=400 Code="InvalidParameter" Message="The specified fault domain count 3 must fall in the range 1 to 2."
```

### Expected Behavior
The azure plugin should be smart enough to figure out what the fault domain count is. Or an API endpoint should be available for the plugin to determine what the proper `platform_update_domain_count` and `platform_fault_domain_count` should be. 

### Actual Behavior
Your code failed, because you didn't know that Azure has limitations, that are semi-poorly referenced here:
https://github.com/MicrosoftDocs/azure-docs/blob/master/includes/managed-disks-common-fault-domain-region-list.md


### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply` (using one of the regions listed above with only 2 Fault Domain

--- a/website/docs/r/availability_set.html.markdown
+++ b/website/docs/r/availability_set.html.markdown
@@ -42,7 +42,11 @@ The following arguments are supported:
 
 * `platform_update_domain_count` - (Optional) Specifies the number of update domains that are used. Defaults to 5.
 
+~> **NOTE:** The number of Update Domains varies depending on which Azure Region you're using - [a list can be found here](https://github.com/MicrosoftDocs/azure-docs/blob/04809e3072e0061ca436b0986cf2222c472651ba/includes/managed-disks-common-fault-domain-region-list.md).
+
 * `platform_fault_domain_count` - (Optional) Specifies the number of fault domains that are used. Defaults to 3.
+
+~> **NOTE:** The number of Fault Domains varies depending on which Azure Region you're using - [a list can be found here](https://github.com/MicrosoftDocs/azure-docs/blob/04809e3072e0061ca436b0986cf2222c472651ba/includes/managed-disks-common-fault-domain-region-list.md).
 
 * `managed` - (Optional) Specifies whether the availability set is managed or not. Possible values are `true` (to specify aligned) or `false` (to specify classic). Default is `false`.
 

--- a/website/docs/r/availability_set.html.markdown
+++ b/website/docs/r/availability_set.html.markdown
@@ -42,11 +42,11 @@ The following arguments are supported:
 
 * `platform_update_domain_count` - (Optional) Specifies the number of update domains that are used. Defaults to 5.
 
-~> **NOTE:** The number of Update Domains varies depending on which Azure Region you're using - [a list can be found here](https://github.com/MicrosoftDocs/azure-docs/blob/04809e3072e0061ca436b0986cf2222c472651ba/includes/managed-disks-common-fault-domain-region-list.md).
+~> **NOTE:** The number of Update Domains varies depending on which Azure Region you're using - [a list can be found here](https://github.com/MicrosoftDocs/azure-docs/blob/master/includes/managed-disks-common-fault-domain-region-list.md).
 
 * `platform_fault_domain_count` - (Optional) Specifies the number of fault domains that are used. Defaults to 3.
 
-~> **NOTE:** The number of Fault Domains varies depending on which Azure Region you're using - [a list can be found here](https://github.com/MicrosoftDocs/azure-docs/blob/04809e3072e0061ca436b0986cf2222c472651ba/includes/managed-disks-common-fault-domain-region-list.md).
+~> **NOTE:** The number of Fault Domains varies depending on which Azure Region you're using - [a list can be found here](https://github.com/MicrosoftDocs/azure-docs/blob/master/includes/managed-disks-common-fault-domain-region-list.md).
 
 * `managed` - (Optional) Specifies whether the availability set is managed or not. Possible values are `true` (to specify aligned) or `false` (to specify classic). Default is `false`.
 

